api/v1alpha1/dnspolicy_types.go

```diff
@@ -215,6 +215,35 @@ func (p *DNSPolicy) DirectReferenceAnnotationName() string {
 	return NewDNSPolicyType().DirectReferenceAnnotationName()
 }
 
+func DNSPolicyStatusMutator(desiredStatus *DNSPolicyStatus, logger logr.Logger) reconcilers.StatusMutatorFunc {
+	return func(obj client.Object) (bool, error) {
+		existing, ok := obj.(*DNSPolicy)
+		if !ok {
+			return false, fmt.Errorf("unsupported object type %T", obj)
+		}
+
+		opts := cmp.Options{
+			cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+			cmpopts.IgnoreMapEntries(func(k string, _ any) bool {
+				return k == "lastTransitionTime"
+			}),
+		}
+
+		if cmp.Equal(*desiredStatus, existing.Status, opts) {
+			return false, nil
+		}
+
+		if logger.V(1).Enabled() {
+			diff := cmp.Diff(*desiredStatus, existing.Status)
+			logger.V(1).Info("status not equal", "difference", diff)
+		}
+
+		existing.Status = *desiredStatus
+
+		return true, nil
+	}
+}
+
```

Comment:
Is this any different from the `v1beta1.KuadrantStatusMutator` and `v1beta2.AuthPolicyStatusMutator`version? Are they likely to ever differ between polices or change between versions of the API?

Reply:
I do not think so. What do you think? Should I make this mutator abstract and reuse for all the types?